editor/src/core/es-modules/package-manager/fetch-packages.ts

```diff
@@ -287,13 +292,31 @@ export async function fetchNodeModules(
   const nodeModulesArr = await Promise.all(
     dependenciesToDownload.map(
       async (newDep): Promise<Either<DependencyFetchError, NodeModules>> => {
+        function notifyFetchEnd(result: ImportOperationResult) {
```

Comment:
an alternaite approach here would be to wrap `fetchNodeModules` into a higher-order function that calls `notifyOperationFinished`. That way `notifyFetchEnd` wouldn't have to be added to each early return

Reply:
I believe the point here is to call `notifyOperationFinished` as and when each of these inner promises complete

Reply:
This inner anon function could however be extracted and wrapped in that way though